asreview/entry_points/auth_tool.py

```diff
@@ -0,0 +1,347 @@
+        "-d",
+        "--db-path",
+        type=str,
+        help="Absolute path to authentication sqlite3 database.",
+        required=True,
+    )
+
+    user_group = user_par.add_mutually_exclusive_group()
+
+    user_group.add_argument(
+        "-i",
+        "--interactive",
+        action="store_true",
+        help="Interactively insert user accounts.",
+    )
+
+    user_group.add_argument(
+        "-j",
+        "--json",
+        type=str,
+        default="[]",
+        help="JSON string that contains a list with user account data.",
+    )
+
+    list_users_par = sub_parser.add_parser(
+        "list-users",
+        help="List user accounts.",
+    )
+
+    list_users_par.add_argument(
+        "-d",
+        "--db-path",
+        type=str,
+        help="Absolute path to authentication sqlite3 database.",
+        required=True,
+    )
+
+    list_projects_par = sub_parser.add_parser(
+        "list-projects",
+        help="List project info from all projects in the ASReview folder.",
+    )
+    list_projects_par.add_argument(
+        "-j",
+        "--json",
+        action="store_true",
+        help="Create JSON string to connect existing projects with users.",
+    )
+
+    link_par = sub_parser.add_parser(
+        "link-projects", help="Link projects to user accounts."
+    )
+
+    link_group = link_par.add_mutually_exclusive_group()
+
+    link_group.add_argument(
+        "-i",
+        "--interactive",
+        action="store_true",
+        help="Interactively link projects to users.",
+    )
+
+    link_group.add_argument(
+        "-j",
+        "--json",
+        type=str,
+        default="[]",
+        help="Use a JSON string to link projects to users.",
+    )
+
+    link_par.add_argument(
+        "-d",
+        "--db-path",
+        type=str,
+        help="Absolute path to authentication sqlite3 database.",
+        required=True,
+    )
+
+    return parser
+
+
+def insert_users(session, entries):
+    """Inserts a list of dictionaries containing user data
+    into the database."""
+    # loop over entries
+    for user in entries:
+        # create a user object
+        user = User(
+            user["email"],
+            email=user["email"],
+            name=user["name"],
+            affiliation=user["affiliation"],
+            password=user["password"],
+            confirmed=True,
+        )
+        try:
+            session.add(user)
+            session.commit()
+            print(f"User with email {user.email} created.")
+        except IntegrityError:
+            print(f"User with identifier {user.email} already exists")
+
+
+def rename_project_folder(project_id, new_project_id):
+    """Rename folder with an authenticated project id"""
+    folder = asreview_path() / project_id
+    folder.rename(asreview_path() / new_project_id)
+    # take care of the id inside the project.json file
+    with open(asreview_path() / new_project_id / "project.json", mode="r") as f:
```

Comment:
Maybe we should put this in a try except block? I don't see why anything would go wrong during reading or writing of `project.json`, but if it does, you already renamed the folder. So something like
```
folder.rename(asreview_path() / new_project_id)
try:
    read_and_write
except Exception:
    folder.rename(asreview_path() / project_id)
```

Reply:
Good call, fixed